components/builder-depot/src/server.rs

```diff
@@ -1241,6 +1241,73 @@ fn list_package_versions(req: &mut Request) -> IronResult<Response> {
     }
 }
 
+fn list_origin_packages_for_version(req: &mut Request) -> IronResult<Response> {
+
+    let (start, stop) = match extract_pagination(req) {
+        Ok(range) => range,
+        Err(response) => return Ok(response),
```

Comment:
This block exists in three other places in this module, so I just followed the existing pattern: it looks like `extract_pagination` early-returns with `Err(iron::response::Response)` on error, and the pattern is to have these handlers return `Ok(the_error)`. Looking around the code at other functions returning `IronResult`, it looks like we _usually_ return `Ok(the_error)` ... but not always. What do you suggest?